### DISCUSSION TITLE
outline (group) data

### DISCUSSION THREAD

**Discussion by jbdemonte** [exceljs/exceljs#1793] [Q&A]
Hi,

I don't understand how outline works.

I'm trying to group columns to get xslx like this (days are grouped on a month total column) 

<img width="1398" alt="Capture d’écran 2021-07-20 à 14 15 08" src="https://user-images.githubusercontent.com/787639/126322139-bbbad753-1dc2-4f75-b086-386b49acb8c4.png">

I don't even know if it is possible with exceljs.

Thanks for you help.

**Comment by Siemienik** [comment]
You have to group and sort the data before generating a file. ExcelJS hasn't go a feature to makes operations like this or execute formulas. 

Some operations can be done by MS Excel when the file is read - for instance, if a formula has set the value to `undefined`.

I will also propose you check [xlsx-renderer](https://github.com/Siemienik/XToolset/tree/master/packages/xlsx-renderer#how-it-works) _(it uses ExcelJS under the hood)_, I see that will help for scenarios like this. Especially the `#! DUMP_COLS` cell type ([Example](https://github.com/Siemienik/XToolset/tree/master/packages/xlsx-renderer/tests/integration/data/Renderer011-DumpCols)).

**Reply by jbdemonte** [reply]
I'm looking for the [+] button in the gray header of the column

![image](https://user-images.githubusercontent.com/787639/129146109-40e1783a-9fca-41ec-8a7c-9e4e676ce99f.png)

I know that sort and data manipulation has to be done by our own, but is there a way to get this column group feature?
It's a visual feature to show / hide those data I guess, not a data manipulation

**Reply by Siemienik** [reply]
I don't know; the easiest way to check it is to make a file with that feature in MS Excel, open it by ExcelJS, and write it again. The feature is maintained if the output is the same as the original. Finally, you will find the proper API  in the source code.  I hope that helps you 😄 

If it isn't maintained, I can do a new feature as an order. Would you mind getting in touch at exjs@siemienik.com?

**Reply by jbdemonte** [reply]
Ok, thanks, I'll test

**Reply by Siemienik** [reply]
@jbdemonte could I ask you about the test results?

**Comment by jbdemonte** [comment]
@Siemienik yes, the feature is still in the copy

I used this code to test it:
```javascript
const ExcelJS = require('exceljs');

(async () => {
  const workbook = new ExcelJS.Workbook();
  
  await workbook.xlsx.readFile('./test.xlsx');
  
  await workbook.xlsx.writeFile('./test2.xlsx');
})()
```

Here is the file : [test.xlsx](https://github.com/exceljs/exceljs/files/7096975/test.xlsx)
